sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileIndexSuite.scala

```diff
@@ -488,6 +489,26 @@ class FileIndexSuite extends SharedSparkSession {
     val fileIndex = new TestInMemoryFileIndex(spark, path)
     assert(fileIndex.leafFileStatuses.toSeq == statuses)
   }
+
+  test("expire FileStatusCache if TTL is configured") {
+    val sparkConfWithTTl = sparkConf.set(SQLConf.METADATA_CACHE_TTL.key, "1")
+
+    withSparkSession(SparkSession.builder.config(sparkConfWithTTl).getOrCreate()) { spark =>
+      val path = new Path("/tmp", "abc")
+      val files = (1 to 3).map(_ => new FileStatus())
+
+      FileStatusCache.resetForTesting()
+      val fileStatusCache = FileStatusCache.getOrCreate(spark)
```

Comment:
plz use `SharedSparkSessionBase.spark` for the session instead.

Reply:
Fixed in https://github.com/apache/spark/pull/28852/commits/ec18c335cd5d2e038766f65b4109e170290afb99